### DISCUSSION TITLE
ZFS pool is not imported during boot on clean-installed OS.

### DISCUSSION THREAD

**Discussion by mikanmi** [openzfs/zfs#13578] [Q&A] [answered]
ZFS pool is **not imported** during boot even though zfs-import-cache has been enabled.
No problem manually importing the ZFS pool by running systemctl restart zfs-import-cache.

```bash
$ zpool list
NAME        SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
root.pool   362G   439M   362G        -         -     0%     0%  1.52x    ONLINE  -

$ sudo systemctl restart zfs-import-cache
$ zpool list
NAME           SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
root.pool      362G   440M   362G        -         -     0%     0%  1.52x    ONLINE  -
storage.pool   928G  75.8G   852G        -         -     0%     8%  1.01x    ONLINE  -

$ sudo reboot
Shared connection to ubuntu.local closed.

$ zpool list
NAME        SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
root.pool   362G   441M   362G        -         -     0%     0%  1.52x    ONLINE  -
```

I clean-installed Ubuntu 22.04 LTS to the LVM volume newly.
I create the _root.pool_ ZFS pool on Ubuntu 22.04 LTS.
I used storage.pool on the old server, Ubuntu 22.04 LTS server.
The problem is that the ZFS file system does **not import** the _storage.pool_ ZFS pool during boot on Ubuntu 22.04 LTS.

Environment:
- Ubuntu 22.04 LTS Server
- ZFS 2.1.2

**Comment by mikanmi** [comment, marked as answer]
I resolve it myself.

**Comment by eviweb** [comment]
Hi @patineboot,
I run in a similar issue.
How did you solve yours, please?
best regards

**Reply by mikanmi** [reply]
See the following page and find your solution.
<https://ounapuu.ee/posts/2021/02/01/how-to-fix-zfs-pool-not-importing-at-boot/>

**Comment by eviweb** [comment]
Hi @patineboot,
thanks for the link.
I found it yesterday but none of the solutions fix my issue.
This is due to a dependency cycle...

**Comment by eviweb** [comment]
Maybe these should interest some one.
My issue occured after an upgrade from Ubuntu 20.04 to 22.04.
The `bpool` ZFS pool was not automatically imported at startup.
I was able to do it manually `zpool import bpool`.
I needed to remove some snapshots as the dataset was full.
The first time, the main issue was due to a missing module, running `depmod` fixed it and allowed me to mount the `/boot/efi` folder.
Then I rebuild my boot loader after a cleanup of noisy old kernels, `grub-install /dev/DEVICE_NAME && update-grub`.
The second time the error came from an usb-c dock connected that failed to mount.
Disconnecting it solved the issue.
